### ISSUE TITLE
[Phase D4] Fused mask+loss Triton kernel to eliminate label clone overhead

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle-legacy#12]
## 背景・目的

現在の `fast_masked_diffusion_loss` は以下の手順で動作する:
1. `masked_labels = labels.clone()` — 全 (B×L) 要素を clone
2. `masked_labels[~diffusion_mask] = -100` — アンマスク位置を上書き
3. `Fast_CrossEntropyLoss.apply(logits, masked_labels, ...)` — Triton CE カーネル

ステップ 1-2 の label clone は `(B, L)` tensor のメモリアロケーションと書き込みを発生させる。これを排除し、`diffusion_mask` を直接 Triton カーネルに渡して1パスで処理する。

## 技術的背景

`unsloth/kernels/cross_entropy_loss.py` の `Fast_CrossEntropyLoss` Triton カーネルは `label == -100` を ignore_index として扱うロジックを持つ。これを `diffusion_mask` の `False` 位置で同様に ignore する形に拡張することで、clone を完全に排除できる。

参考: LLaDA の損失計算は `(masked_label != -100) & diffusion_mask` のような組み合わせを使う。

## Acceptance Criteria

- [ ] `unturtle/kernels/fused_masked_diffusion_loss.py` に `fused_masked_diffusion_loss` を実装
  - 引数: `logits (B,L,V)`, `labels (B,L)`, `diffusion_mask (B,L) bool`, `loss_weights (B,) or None`
  - label clone なし、diffusion_mask を Triton カーネル内で直接参照
  - 数値結果が既存 `fast_masked_diffusion_loss` と bit-exact または十分に近い (atol=1e-5)
- [ ] `fast_masked_diffusion_loss` の内部で GPU 時は `fused_masked_diffusion_loss` を使用するよう切り替え
- [ ] ベンチマーク: `dev/benchmark_loss.py` を更新し fused vs 現行を比較
  - 期待: B=4, L=512, V=128K で clone 排除による ~10-20% 高速化

## テスト
- `tests/diffusion/test_masked_diffusion_loss.py` に fused kernel の数値正確性テスト追加
- GPU accuracy テスト (`tests/diffusion/test_gpu_accuracy.py`) で fused path も検証

## 関連ファイル
- `unturtle/kernels/masked_diffusion_loss.py` — 現行実装
- `unsloth/kernels/cross_entropy_loss.py` — 参照 CE Triton カーネル

## 優先度
Low — Phase D3 (sequence packing) よりも後回し可。ただし vocab が大きい (128K+) モデルでは効果が大きい。
